lib/Models/Terria.ts

```diff
@@ -759,23 +760,73 @@ export default class Terria {
   }
 
   protected async forceLoadInitSources(): Promise<void> {
+    const loadInitSource = createTransformer(
+      async (initSource: InitSource): Promise<JsonObject | undefined> => {
+        let jsonValue: JsonValue | undefined;
+        if (isInitUrl(initSource)) {
+          try {
+            jsonValue = await loadJson5(initSource.initUrl);
+          } catch (e) {
+            raiseErrorToUser(
+              this,
+              new TerriaError({
+                sender: this,
+                title: { key: "models.terria.loadingInitSourceErrorTitle" },
+                message: {
+                  key: "models.terria.loadingInitSourceError2Message",
+                  parameters: { loadSource: initSource.initUrl }
+                }
+              })
+            );
+          }
+        } else if (isInitOptions(initSource)) {
+          let error: any;
+          for (const option of initSource.options) {
+            try {
+              jsonValue = await loadInitSource(option);
+              if (jsonValue !== undefined) break;
+            } catch (err) {
+              error = err;
+            }
+          }
+          if (jsonValue === undefined && error !== undefined) throw error;
+        } else if (isInitData(initSource)) {
+          jsonValue = initSource.data;
+        } else if (isInitDataPromise(initSource)) {
+          jsonValue = (await initSource).data;
+        }
+
+        if (jsonValue && isJsonObject(jsonValue)) {
+          return jsonValue;
+        }
+        return undefined;
+      }
+    );
+
```

Comment:
This has been moved here from outside `Terria` class - because I wanted access to instance variable for `raiseErrorToUser`